datafusion/expr/src/udf.rs

```diff
@@ -216,6 +217,15 @@ impl ScalarUDF {
         self.inner.is_nullable(args, schema)
     }
 
+    /// Returns the field metadata for this function.
+    pub fn metadata(
```

Comment:
This feels similar in usecase to https://github.com/apache/datafusion/pull/13290 -- if we merged https://github.com/apache/datafusion/pull/13290 would that solve your usecase?

Reply:
Looks like not. This pr is for supporting custom metadata when converting scalar function expr to schema field during planing.